### PR TITLE
refactor(GatewayGuildCreateDispatchData): Narrow `channels` and `threads` types

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**
- `channels` will not contain direct message channels or group direct message channels. These are channels attached to a guild.
- `threads` are, well, threads. This cannot contain category channels, for example.

**If applicable, please reference Discord API Docs PRs or commits that influenced this PR:**
N/A

### CODE REVIEW

File: deno/gateway/v10.ts [hunk 1]
```diff
@@ -879,15 +879,17 @@ export interface GatewayGuildCreateDispatchData extends APIGuild {
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	channels: APIChannel[];
+	channels: (APIChannel & { type: Exclude<ChannelType, ChannelType.DM | ChannelType.GroupDM> })[];
```
Comment (sdanialraza) [inline deno/gateway/v10.ts:882]: [GuildChannelType](https://github.com/discordjs/discord-api-types/blob/main/payloads/v10/channel.ts#L55) can be used here

File: deno/gateway/v10.ts [hunk 2]
```diff
@@ -879,15 +879,17 @@ export interface GatewayGuildCreateDispatchData extends APIGuild {
...
 	/**
 	 * Threads in the guild
 	 *
 	 * **This field is only sent within the [GUILD_CREATE](https://discord.com/developers/docs/topics/gateway-events#guild-create) event**
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	threads: APIChannel[];
+	threads: (APIChannel & {
+		type: ChannelType.AnnouncementThread | ChannelType.PrivateThread | ChannelType.PublicThread;
```
Comment (sdanialraza) [inline deno/gateway/v10.ts:891]: [ThreadChannelType](https://github.com/discordjs/discord-api-types/blob/main/payloads/v10/channel.ts#L208) can be used here

**Review by sdanialraza — CHANGES_REQUESTED** [review]

**Comment by Jiralite** [comment]
Thank you for that! Should be resolved with f7d03fe63f228326cc08d7db1770a09e7990d8fe.

**Review by sdanialraza — APPROVED** [review]

File: gateway/v10.ts [hunk 3]
```diff
@@ -879,15 +882,15 @@ export interface GatewayGuildCreateDispatchData extends APIGuild {
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	channels: APIChannel[];
+	channels: APIGuildChannel<Exclude<GuildChannelType, ThreadChannelType>>[];
```
[the remark this answers is not in the document]
Comment (Jiralite) [reply gateway/v10.ts]: I wasn't sure which to use. More concrete types sounds better so I'll switch back to this.

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**